### PR TITLE
Support bindings with the Super (Meta/Cmd/Win) modifier

### PR DESCRIPTION
**Pull request by the-mikedavis**
Terminals which support the enhanced keyboard protocol send events for keys pressed with the Super modifier (Windows/Linux key or the Command key). We can support keybindings with the Super modifier in Helix with a few small changes: mapping the event from crossterm and parsing/representing the modifier in the KeyEvent string representation.

The binding can be done with `Meta-`, `Win-` or `Cmd-` prefixes - these are all synonymous. For example `keys.normal.Cmd-s = ":write"`.

Closes https://github.com/helix-editor/helix/issues/2913
Closes https://github.com/helix-editor/helix/issues/6582

### CODE REVIEW

File: helix-view/src/input.rs
Comment (the-mikedavis) [inline helix-view/src/input.rs:163]: We could use another name/letter for this in the representation. Either [Meta](https://en.wikipedia.org/wiki/Meta_key) or Hyper would probably be close enough but would be incorrect. Those two are also possible to receive as distinct modifiers although it's unlikely (https://sw.kovidgoyal.net/kitty/keyboard-protocol/#modifiers)

Comment (eugenesvk) [reply helix-view/src/input.rs:163]: maybe `W-` as it's a single letter abbreviation that's closer to Super (Win-key) and also correct unlike Meta/Hyper?

Comment (eugenesvk) [reply helix-view/src/input.rs:163]: Well, `Super` is incorrect everywhere as it has a different name on each OS and given that the first letters aren't unique, first-letter OS-agnostic-but-correct abbreviations aren't an option (or maybe use two-letter one? `CW-` cmd-win)

Comment (the-mikedavis) [reply helix-view/src/input.rs:163]: It seems to be a common choice for free software, apparently because of trademark issues: https://en.wikipedia.org/wiki/Super_key_(keyboard_button)#Usage

`OS-` is could be a good choice for this (https://support.system76.com/articles/pop-keyboard-shortcuts/). QMK calls it GUI but I think that's a little confusing (https://docs.qmk.fm/#/keycodes_basic?id=modifiers). I don't like `CW-`: it isn't used anywhere else as far as I know.

Comment (jpsecher) [reply helix-view/src/input.rs:163]: So how do we reach a decision on this?  I think it would be wise to avoid inventing yet another name for the key, and if we want to avoid something that starts with `s`, then the choices seem to be

1. `win`
2. `windows`
3. `cmd`
4. `command`
5. `gui`
6. `meta`
7. `os`

Comment (realeinherjar) [reply helix-view/src/input.rs:163]: I like `OS-`

**Comment by koalalorenzo** [comment]
Any plans for this PR? :)

File: helix-view/src/input.rs
Comment (matta) [reply helix-view/src/input.rs:163]: I did some digging on this, probably too much.

The world already uses inconsistent terms for the Windows/Command key in cross-platform contexts, but it seems to *consistently* pick between either "Meta" or "Super".

I don't think it would be good to invent a new name for the key. Helix should pick "Meta" or "Super". I'd pick "Meta" but either choice is reasonable.

From what I can suss out, "Meta" is the most widely used. It is what web standards call it: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/metaKey, and it is probably what most people will readily understand. I say "most people" because I'm guessing web developers may outnumber basically all other kinds of developers, and web apps outnumber basically all other kinds of apps (at least those with any kind of broad reach), so web standards win any familiarity context. E.g. VS Code is the most popular "free" editor in the world, probably the most popular editor full stop, and it calls it "Meta," because it is basically a browser app.

(Interestingly, I fired up RustRover, and it doesn't support binding the Windows key, at least on my Gnome box.)

Calling it "Super" would be consistent with X11, Gnome, and the Kitty protocol at https://sw.kovidgoyal.net/kitty/keyboard-protocol/#modifiers.

Calling it "Meta" would be consistent with the web, and KDE, and the Linux kernel, but may confuse die hard GNU Readline and Emacs users, and other X11-first software.

I'd pick "Meta" because the influence of web standards on developer mind share is hard to ignore.

Comment (eugenesvk) [reply helix-view/src/input.rs:163]: > and web apps outnumber basically all other kinds of apps (at least those with any kind of broad reach), so web standards win any familiarity context. E.g. VS Code is the most popular "free" editor in the world, probably the most popular editor full stop, and it calls it "Meta," because it is basically a browser app.

But broader user base outnumbers any specific tech-stack user base by a wide margin, so using an obscure tech-stack-based name can't be a sensible alternative to using a widely known name

And your VSCode example confirms this since VSCode doesn't use Meta on the platforms where it's unknown (Mac and Windows) in it's user-facing config (even though it does use the web-standards Meta name in debugigng output): 

> [Accepted keys](https://code.visualstudio.com/docs/getstarted/keybindings#_accepted-keys)
The key is made up of modifiers and the key itself.

>  The following modifiers are accepted:

>  Platform	Modifiers
macOS	Ctrl+, Shift+, Alt+, Cmd+
Windows	Ctrl+, Shift+, Alt+, Win+
Linux	Ctrl+, Shift+, Alt+, Meta+

By the way, why not allow any value from `Cmd`/`Win`/`Meta` and simply treat either config value as `Cmd` on a Mac,`Win` on Windows, and the Linux key alternative on Linux

Comment (the-mikedavis) [reply helix-view/src/input.rs:163]: I like the idea of Cmd+Win+Meta synonyms! Let's go with that

**Comment by mighabana** [comment]
Hi is there any way I can help to get this merged? I'd be happy to help make any code changes or have any design descussions if that's what's blocking the PR. 😄

**Comment by the-mikedavis** [comment]
It just needs a round of review from the other maintainers, codewise it should  be fine

**Review by pascalkuthe — APPROVED** [review]

**Comment by omentic** [comment]
I have no stake in this PR, but: why all of `Meta-`, `Cmd-`, `Win-`?

It seems simpler to pick just one (or to pick `M-`, on analogy with `S-`, `A-`, `C-`)

**Comment by the-mikedavis** [comment]
See the collapsed thread above - it started as "Super-" but there isn't an ideal choice for what to call the modifier

**Comment by NewtonChutney** [comment]
Hello!
Have any links pointing to terminals supporting `enhanced keyboard protocol` been added in the docs?

**Pull request merged**
